### ISSUE TITLE
5.3 Can't use multiple connection while they are the same driver

### ISSUE THREAD

**Issue by HanSon** [laravel/framework#17747]
I'm using two connections, both of them are mysql, and I found out no matter how it always query the first one , which is the default one .

And I look up the ConnectionFactory code, and there might have some problem

```
protected function createConnection($driver, $connection, $database, $prefix = '', array $config = [])
    {
        if ($this->container->bound($key = "db.connection.{$driver}")) {
            return $this->container->make($key, [$connection, $database, $prefix, $config]);
        }

        switch ($driver) {
            case 'mysql':
                return new MySqlConnection($connection, $database, $prefix, $config);
            case 'pgsql':
                return new PostgresConnection($connection, $database, $prefix, $config);
            case 'sqlite':
                return new SQLiteConnection($connection, $database, $prefix, $config);
            case 'sqlsrv':
                return new SqlServerConnection($connection, $database, $prefix, $config);
        }

        throw new InvalidArgumentException("Unsupported driver [$driver]");
    }
```

When the first connection is make by the container, the second one will be the same as the first one

```
public function make($abstract, array $parameters = [])
    {
        $abstract = $this->getAlias($this->normalize($abstract));

        if (isset($this->instances[$abstract])) {
           // second one will go here
            return $this->instances[$abstract];
        }
    ...
  }
```

So , when I try to use `DB::connection('mysql2')->getName()`, it will return mysql1 intead of mysql2

see [PR #17748 ](https://github.com/laravel/framework/pull/17748)

**Comment by themsaid** [comment]
Please upgrade to 5.4 and check if the problem still exists.

**Comment by HanSon** [comment]
5.4 fixed this problem but I think 5.3 should fix this

**Comment by kieranheadley** [comment]
Is there a fix for this without upgrading to 5.4?

**Comment by GrahamCampbell** [comment]
Unforutnately, it's not standard practice to fix bugs in non-supported versions. Please upgrade to Laravel 5.5 LTS as soon as possible.
